javascript/ql/src/semmle/javascript/security/dataflow/DomBasedXss.qll

```diff
@@ -91,7 +91,8 @@ module DomBasedXss {
           isPrefixOfJQueryHtmlString(astNode, prefix) and
           strval = prefix.asExpr().getStringValue() and
           not strval.regexpMatch("\\s*<.*")
-        )
+        ) and
+        not isDocumentURL(astNode)
```

Comment:
could we restrict this to only whitelist the `$`-calls?

` not (call = jquery().getACall() and isDocumentURL(astNode))`

Reply:
It's not very obvious, but I believe we already have restriction you are asking for.

See the first half of the disjunction a few lines above.
The second half of the `or` is only relevant when the input is primarily interpreted as a selector, that is, it has to start with `<` for the value to be interpreted as HTML. In practice means it's an argument to `$()`.

I've added a test case for it just to be sure.